### PR TITLE
integrate common test with riak_test

### PR DESCRIPTION
**Pull request by gordonguthrie**

### CODE REVIEW

**Comment by javajolt** [comment]
All the TS, riak_test tests pass. I'll try @andytill's flexible keys one now.

File: src/riak_test_escript.erl [hunk 1]
```diff
@@ -429,3 +444,13 @@ so_kill_riak_maybe() ->
             io:format("Leaving Riak Up... "),
             rt:whats_up()
     end.
+
+is_common_test([]) -> 
+    false;
+is_common_test([{attribute, _, file, {File, _}} | T]) ->
+    case filename:basename(File) of
```
Comment (tburghart) [inline src/riak_test_escript.erl]: It seems there must be a better way to test this, isn't there?

Comment (javajolt) [inline src/riak_test_escript.erl]: Do you have a suggestion, @tburghart?

Comment (andytill) [inline src/riak_test_escript.erl]: rebar requires all common tests end in `_SUITE.erl`.

Comment (javajolt) [inline src/riak_test_escript.erl]: @andytill Is that true of rebar3, too?

Comment (andytill) [inline src/riak_test_escript.erl]: No idea, I'm an erlang.mk guy ;)

rebar2 at least required the module name without the SUITE and then add it to the end to form the file name.

File: src/riak_test_escript.erl [hunk 2]
```diff
@@ -429,3 +444,13 @@ so_kill_riak_maybe() ->
...
+        "ct.hrl" -> true;
```
Comment (hmmr) [inline src/riak_test_escript.erl]: Will this work if the source has `-include_lib("common_test/include/ct.hrl").`?

**Comment by javajolt** [comment]
I tried three of @andytill's tests from his CT PR:
- ts_cluster_keys_SUITE
- ts_cluster_quantum_boundaries_SUITE
- ts_updown_select_SUITE

The first two passed, but the third failed.  I suspect it need some other changes, to pass, however.

**Comment by andytill** [comment]
@javajolt `ts_updown_select_SUITE.erl` was just a test suite for developing the common test integration, it is just a copy/pasta of one of the other basic tests. It should be removed.  Shall I do this here or does it need to go on another branch?

**Comment by javajolt** [comment]
OK, if that is a red herring, then let's get this thing merged.

**Comment by javajolt** [comment]
:+1:

**Pull request merged**